### PR TITLE
add C binary_search and linear_search

### PR DESCRIPTION
**Pull request by Guanngxu**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by Guanngxu** [comment]
image

**Comment by Guanngxu** [comment]
image

**Comment by Gonglja** [comment]
Thanks @Guanngxu

**Review by Gonglja — APPROVED** [review]

File: codes/c/chapter_searching/linear_search.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/* 线性查找（链表） */
+ListNode* linearSearchLinkedList(ListNode* head, int target) {
+    // 遍历链表
+    while (head != NULL) {
+        // 找到目标结点，返回之
+        if (head->val == target)
+            return head;
+        head = head->next;
+    }
+    // 未找到目标结点，返回 nullptr
```
Comment (krahets) [inline codes/c/chapter_searching/linear_search.c:30]: NULL

**Pull request merged**